src/_posts/2018-08-XX-review-input-streaming-connectors.md

```diff
@@ -0,0 +1,224 @@
+
+Below are the main streaming input connectors for available for Beam and Spark DStreams in Java:
+
+<table>
+  <tr>
+   <td>
+   </td>
+   <td>
+   </td>
+   <td><strong>Apache Beam</strong>
+   </td>
+   <td><strong>Apache Spark DStreams</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >File Systems
+   </td>
+   <td>Local<br>(Using the <code>file://</code> URI)
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/TextIO.html">TextIO</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/streaming/StreamingContext.html#textFileStream-java.lang.String-">textFileStream</a><br>(Spark treats most Unix systems as HDFS-compatible, but the location should be accessible from all nodes)
+   </td>
+  </tr>
+  <tr>
+   <td>HDFS<br>(Using the <code>hdfs://</code> URI)
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/hdfs/HadoopFileSystemOptions.html">HadoopFileSystemOptions</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/streaming/util/HdfsUtils.html">HdfsUtils</a>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >Object Stores
+   </td>
+   <td>Cloud Storage<br>(Using the <code>gs://</code> URI)
+   </td>
+   <td rowspan="2" ><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/hdfs/HadoopFileSystemOptions.html">HadoopFileSystemOptions</a>
+   </td>
+   <td rowspan="2" ><a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/SparkContext.html#hadoopConfiguration--">hadoopConfiguration</a>
+<p>
+and <a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/streaming/StreamingContext.html#textFileStream-java.lang.String-">textFileStream</a>
+   </td>
+  </tr>
+  <tr>
+   <td>S3<br>(Using the <code>s3://</code> URI)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >Messaging Queues
+   </td>
+   <td>Kafka
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/kafka/KafkaIO.html">KafkaIO</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/streaming-kafka-0-10-integration.html">spark-streaming-kafka</a>
+   </td>
+  </tr>
+  <tr>
+   <td>Kinesis
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/kinesis/KinesisIO.html">KinesisIO</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/streaming-kinesis-integration.html">spark-streaming-kinesis</a>
+   </td>
+  </tr>
+  <tr>
+   <td>Cloud Pub/Sub
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.html">PubsubIO</a>
+   </td>
+   <td><a href="https://github.com/apache/bahir/tree/master/streaming-pubsub">Spark-streaming-pubsub</a> from <a href="http://bahir.apache.org">Apache Bahir</a>
```

Comment:
Other streaming connectors seem to be missing for both Beam and spark (not sure if they are not because those are not for distributed data stores but could make the comparison richer): JMS, MQTT, AMQP

Reply:
Which connectors exactly would you recommend in each case for Spark? Bahir has one for MQTT, but I'm not sure what other connectors to recommend for JMS & AMQP.

Reply:
There seems not to be community maintanied version of both so probably just mentioned that.